File: examples/minimal_flowgraph.cpp
#include <fmt/core.h>
#include <gnuradio-4.0/Graph.hpp>
#include <gnuradio-4.0/Scheduler.hpp>
#include <gnuradio-4.0/packet-modem/null_sink.hpp>
#include <gnuradio-4.0/packet-modem/null_source.hpp>
#include <boost/ut.hpp>

int main()
{
    using namespace boost::ut;

    gr::Graph fg;
    auto& source = fg.emplaceBlock<gr::packet_modem::NullSource<int>>();
    auto& sink = fg.emplaceBlock<gr::packet_modem::NullSink<int>>();
    expect(eq(gr::ConnectionResult::SUCCESS, fg.connect<"out">(source).to<"in">(sink)));

    gr::scheduler::Simple sched{ std::move(fg) };
    expect(sched.runAndWait().has_value());

    return 0;
}

File: python/examples/minimal_flowgraph.py
#!/usr/bin/env python3

import gr4_packet_modem

sched = gr4_packet_modem.SchedulerSimpleSingleThreaded()
fg = sched.graph()

type_ = 'float'
source = fg.emplaceBlock('gr::packet_modem::NullSource', type_, {})
head = fg.emplaceBlock(
    'gr::packet_modem::Head', type_,
    {'num_items': gr4_packet_modem.pmt_from_uint64_t(100_000_000_000)})
sink = fg.emplaceBlock('gr::packet_modem::NullSink', type_, {})

fg.connect(source, 'out', head, 'in')
fg.connect(head, 'out', sink, 'in')

sched.runAndWait()
